include/firefly/intel64/cpu/cpu.hpp

```diff
@@ -17,4 +18,27 @@ struct CpuData {
 
 void initializeThisCpu(uint64_t stack);
 CpuData &thisCpu();
+
+// from linux kernel source
+static inline void native_cpuid(unsigned int *eax, unsigned int *ebx,
+                                unsigned int *ecx, unsigned int *edx)
+{
+        /* ecx is often an input as well as an output. */
+        asm volatile("cpuid"
+            : "=a" (*eax),
+              "=b" (*ebx),
+              "=c" (*ecx),
+              "=d" (*edx)
+            : "0" (*eax), "2" (*ecx));
+}
+
+static inline bool cpu_support_onegb_pages() {
```

Comment:
Again, we have no explicit coding guidelines yet but function names should be camel case (yes, I'm aware that I violated this myself in a few places. I'll have to refactor all that stuff)